keras_core/backend/tensorflow/distribution_lib.py

```diff
@@ -0,0 +1,65 @@
+"""!!!DO NOT USE!!!
+
+Distribution related class for Tensorflow backend.
+
+This is just a prototype and we might want to unify it
+with other backends in the future.
+"""
+import tensorflow as tf
+from tensorflow.experimental import dtensor
+
+
+def list_devices(device_type=None):
+    """Return all the available devices based on the device type.
+
+    Note that this should return the global devices in a distributed setting.
+
+    Args:
+        device_type: string of `"cpu"`, `"gpu"` or `"tpu"`. Default to `gpu` or
+        `tpu` if available when device_type is not provided. Otherwise will
+        return the `cpu` devices.
+
+    Return:
+        List of devices that are available for distribute computation.
+    """
+    device_type = (
+        device_type.upper() if device_type else dtensor.preferred_device_type()
+    )
+    return tf.config.list_logical_devices(device_type=device_type)
+
+
+def to_dtensor_mesh(device_mesh):
+    """Convert the DeviceMesh to Tensorflow backend specific Mesh.
+
+    Args:
+        device_mesh: DeviceMesh instance to convert.
+
+    Returns:
+        A `tf.dtensor.Mesh` instance.
+    """
+    mesh_dims = list(zip(device_mesh.axis_names, device_mesh.shape))
+    return dtensor.create_mesh(
```

Comment:
I think the usage of create_mesh will limit it to be a single worker mesh. We might want to use the create_distributed_mesh https://www.tensorflow.org/api_docs/python/tf/experimental/dtensor/create_distributed_mesh